src/formatter/html.ts

```diff
@@ -0,0 +1,286 @@
+		const warningsText = quiet ? "" : `, ${totalWarningCount} ${totalWarningCount === 1 ? "warning" : "warnings"}`;
+		const problemsText = `${totalCount} ${totalCount === 1 ? "problem" : "problems"}`;
+
+		const summary = `<div class="summary">
+			<h2>Summary</h2>
+			<p>
+				${problemsText} (${errorsText}${warningsText})
+			</p>
+			${totalFatalErrorCount ? `<p><strong>${totalFatalErrorCount} fatal errors</strong></p>` : ""}
+			${!autofix && totalCount > 0 ?
+				"<p>Run <code>ui5lint --fix</code> to resolve all auto-fixable problems</p>" :
+				""}
+		</div>`;
+
+		// Full HTML document with some basic styling
+		const html = `<!DOCTYPE html>
+<html lang="en">
+<head>
+	<meta charset="UTF-8">
+	<meta name="viewport" content="width=device-width, initial-scale=1.0">
+	<title>UI5 Linter Report</title>
+	<style>
+		body {
+			font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto,
+				Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
+			line-height: 1.5;
+			max-width: 1200px;
+			margin: 0 auto;
+			padding: 20px;
+			color: #333;
+		}
+		h1, h2 {
+			border-bottom: 1px solid #eaecef;
+			padding-bottom: 0.3em;
+			color: #24292e;
+		}
+		h3 {
+			padding: 10px;
+			margin: 0;
+			background-color: #f6f8fa;
+			border-top-left-radius: 4px;
+			border-top-right-radius: 4px;
+			border: 1px solid #eaecef;
+			border-bottom: none;
+		}
+		table {
+			width: 100%;
+			border-collapse: collapse;
+			margin-bottom: 20px;
+			border: 1px solid #eaecef;
+			border-radius: 4px;
+		}
+		th, td {
+			text-align: left;
+			padding: 8px 12px;
+			border-bottom: 1px solid #eaecef;
+		}
+		th {
+			background-color: #f6f8fa;
+			font-weight: 600;
+		}
+		tr.error {
+			background-color: #fff5f5;
+		}
+		tr.error td:first-child {
+			color: #d73a49;
+			font-weight: 600;
+		}
+		tr.warning {
+			background-color: #fffbea;
+		}
+		tr.warning td:first-child {
+			color: #e36209;
+			font-weight: 600;
+		}
+		tr.fatal-error {
+			background-color: #ffdce0;
+		}
+		tr.fatal-error td:first-child {
+			color: #b31d28;
+			font-weight: 600;
+		}
+		code {
+			background-color: #f6f8fa;
+			padding: 0.2em 0.4em;
+			border-radius: 3px;
+			font-family: SFMono-Regular, Consolas, 'Liberation Mono', Menlo, monospace;
+		}
+		.summary {
+			margin-bottom: 30px;
+		}
+		.file {
+			margin-bottom: 30px;
+		}
+		.note {
+			margin-top: 20px;
+			padding: 10px;
+			background-color: #f6f8fa;
+			border-radius: 4px;
+			font-size: 14px;
+		}
+		a {
+			color: #0366d6;
+			text-decoration: none;
+		}
+		a:hover {
+			text-decoration: underline;
+		}
+		@media (max-width: 768px) {
+			body {
+				padding: 10px;
+			}
+			table {
+				display: block;
+				overflow-x: auto;
+			}
+		}
+	</style>
+</head>
+<body>
+	<h1>UI5 Linter Report</h1>
+	<p>Generated with UI5 Linter v${version}</p>
+
+	${summary}
+
+	${resultsHtml ? `<h2>Findings</h2>${resultsHtml}` : "<p>No issues found. Your code looks great!</p>"}
```

Comment:
I really like the simplicity of your HTML styling and its consistency to the markdown format. However, to be a bit more consistent and to keep the simple and informative shape, I'd suggest to remove the paragraph element.
```suggestion
	${resultsHtml && `<h2>Findings</h2>${resultsHtml}`}
```